pkg/server/config.go

```diff
@@ -245,35 +256,34 @@ func GetTotalMemory() (int64, error) {
 				log.Infof(context.TODO(), "can't read available memory from cgroups (%s), using system memory %s instead", err,
 					humanizeutil.IBytes(totalMem))
 			}
-			return totalMem, nil
+			return checkTotal(totalMem)
 		}
 		var cgAvlMem uint64
 		if cgAvlMem, err = strconv.ParseUint(strings.TrimSpace(string(buf)), 10, 64); err != nil {
 			if log.V(1) {
 				log.Infof(context.TODO(), "can't parse available memory from cgroups (%s), using system memory %s instead", err,
 					humanizeutil.IBytes(totalMem))
 			}
-			return totalMem, nil
+			return checkTotal(totalMem)
 		}
 		if cgAvlMem > math.MaxInt64 {
 			if log.V(1) {
 				log.Infof(context.TODO(), "available memory from cgroups is too large and unsupported %s using system memory %s instead",
```

Comment:
the first `%s` is outside of the flow of the sentence. It would look better in parentheses, or moved up to read "available memory %s from cgroups is too large, ...". Still only a nit.